File: VB/Core/CommonVB.Core/Extensions/ConfigurationExtension.vb
Imports System.IO
Imports System.Runtime.CompilerServices
Imports Microsoft.Extensions.Configuration

Public Module ConfigurationExtension

    <Extension>
    Public Function Initialize(builder As IConfigurationBuilder) As IConfigurationBuilder

        Dim env As String = Environment.GetEnvironmentVariable("DOTNET_ENVIRONMENT")
        If String.IsNullOrWhiteSpace(env) Then
            env = "Production"
        End If

        Return New ConfigurationBuilder() _
                .SetBasePath(Directory.GetCurrentDirectory()) _
                .AddJsonFile("appsettings.json", optional:=True, reloadOnChange:=True) _
                .AddJsonFile($"appsettings.{env}.json", optional:=True, reloadOnChange:=True) _
                .AddEnvironmentVariables()

    End Function

End Module

File: CSharp/Core/Common.Core/Extensions/ConfigurationExtension.cs
using Microsoft.Extensions.Configuration;

namespace Common.Core.Extensions;

public static class ConfigurationExtension
{
    public static IConfigurationBuilder Initialize(this IConfigurationBuilder builder)
    {
        string env = Environment.GetEnvironmentVariable("DOTNET_ENVIRONMENT") ?? "Production";

        return builder
            .SetBasePath(Directory.GetCurrentDirectory())
            .AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
            .AddJsonFile($"appsettings.{env}.json", optional: true, reloadOnChange: true)
            .AddEnvironmentVariables();
    }
}